src/hotspot/share/runtime/thread.hpp

```diff
@@ -958,8 +958,8 @@ class JavaThread: public Thread {
   jlong*    _jvmci_counters;
 
   // Fast thread locals for use by JVMCI
-  intptr_t*  _jvmci_reserved0;
-  intptr_t*  _jvmci_reserved1;
+  jlong      _jvmci_reserved0;
```

Comment:
We are trying to avoid use of Java types in code that does not directly interact with Java. Does this really need to be a jlong or does it just need to be 64-bit? Or does it need to be the same size as a ptr whatever that may be?

Reply:
Yes that's right.  Since it's now exposed through Java it really needs to be a Java compatible type.

Reply:
I had that same thought but I agree with Tom.  We want a Java type here.